ParkingFinder/repositories/real_time_repository.py

```diff
@@ -0,0 +1,97 @@
+from sqlalchemy.orm.exc import NoResultFound
+from tornado.gen import coroutine, Return
+
+from ParkingFinder.base.async_db import create_session
+from ParkingFinder.base.errors import NotFound
+from ParkingFinder.mappers.real_time_mapper import RealTimeMapper
+from ParkingFinder.tables.real_time import RealTimes
+
+
+class RealTimeRepository(object):
+
+    @staticmethod
+    @coroutine
+    def read_one(user_id):
+        """
+        Read from realtime with user_id 
+
+        :param str key:
+        :return data:
+        :raises sqlalchemy.orm.exc.NoResultFound:
+        """
+        with create_session() as session:
+            realtime = session.query(RealTimes).filter(
+                RealTimes.waiting_user_id == user_id
+            ).one()
+            entity = RealTimeMapper.to_entity(record=realtime)
+            raise Return(entity)
+
+    @classmethod
+    @coroutine
+    def upsert(cls, user_id, realtime):
+        """
+        Update/Insert realtime, if realtime exist, update realtime's information
+        otherwise, insert a new realtime, and realtime have to be a realtime entity
+        with all required fields
+
+        :param RealTime realtime:
+        :return RealTime:
+        :raises AssertionError: the user provided doesn't have user_id
+        :raises ValidationError: the user provided missing required fields
+        """
+
+        try:
+            _realtime = yield cls._update(user_id, realtime)
+            raise Return(_realtime)
+
+        except NoResultFound:
+            _realtime = yield cls._insert(realtime)
+            raise Return(_realtime)
+
+    @classmethod
+    @coroutine
+    def _update(cls,
+                user_id,
+                realtime
+                ):
+        """
+
+        :param str user_id:
+        :param Realtime realtime:
+        :return RealTime: updated RealTime
+        :raises NoResultFound: the realtime that is requested to update is not exist
+        """
+
+        with create_session() as session:
+            _realtime = session.query(RealTimes).filter(
+                RealTimes.waiting_user_id == realtime.waiting_user_id
```

Comment:
Here, you are update the location of the one send the request and ignore the other.